test/widgets/action_sheet_test.dart

```diff
@@ -1188,6 +1189,199 @@ void main() {
       });
     });
   });
+
+  group('channel action sheet', () {
+    late ZulipStream someChannel;
+    const someTopic = 'my topic';
+    late StreamMessage someMessage;
+
+    Future<void> prepare({UnreadMessagesSnapshot? unreadMsgs}) async {
+      final stream = eg.stream();
+      someChannel = stream;
+      someMessage = eg.streamMessage(
+      stream: someChannel, topic: someTopic, sender: eg.otherUser);
+      addTearDown(testBinding.reset);
+
+      unreadMsgs ??= eg.unreadMsgs(channels: [
+        eg.unreadChannelMsgs(
+          streamId: stream.streamId,
+          topic: 'topic',
+          unreadMessageIds: [1],
+        ),
+      ]);
+
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        realmUsers: [eg.selfUser, eg.otherUser],
+        streams: [someChannel],
+        subscriptions: [eg.subscription(someChannel)],
+        unreadMsgs: unreadMsgs));
+      store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
```

Comment:
A good way to keep the test data internally consistent, in order to help keep it realistic, is to make less use of arguments to `eg.initialSnapshot`, and instead add add the messages, users, etc., through events. The snapshot has a lot of internal invariants that the server takes care of when producing it, and if you generate a snapshot you're responsible for upholding those, but by handling events it becomes our model code's responsibility to maintain all the invariants.

Try `store.addMessage`, and the other methods defined next to that one.